Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -86,8 +87,24 @@ public object Create(IonSerializationOptions options, IIonReader reader, Type ta
             if (annotations.Length > 0)
             {
                 var typeName = annotations[0];
-                var assemblyName = options.AnnotatedTypeAssemblies.First(a => Type.GetType(FullName(typeName, a)) != null);
-                return Activator.CreateInstance(Type.GetType(FullName(typeName, assemblyName)));
+                Type typeToCreate = null;
+                try
+                {
+                    var assemblyName = options.AnnotatedTypeAssemblies.First(a => Type.GetType(FullName(typeName, a)) != null);
+                    typeToCreate = Type.GetType(FullName(typeName, assemblyName));
+                }
+                catch

```

Comment:
Is there an `if-else` version of this? Control flow via exceptions, especially, just *any* exception, is less deterministic.

Reply:
Updated to catch the specific exception when `First` does not find any matching elements.